ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,816 @@
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	remoteManagement, err := getItemPriceId(items, "remote_management", "REBOOT_KVM_OVER_IP")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vpnManagement, err := getItemPriceId(items, "vpn_management", "UNLIMITED_SSL_VPN_USERS_1_PPTP_VPN_USER_PER_ACCOUNT")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	notification, err := getItemPriceId(items, "notification", "NOTIFICATION_EMAIL_AND_TICKET")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	response, err := getItemPriceId(items, "response", "AUTOMATED_NOTIFICATION")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vulnerabilityScanner, err := getItemPriceId(items, "vulnerability_scanner", "NESSUS_VULNERABILITY_ASSESSMENT_REPORTING")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	// Define an order object using basic paramters.
+
+	order := datatypes.Container_Product_Order{
+		ContainerIdentifier: sl.String(d.Get("hostname").(string)),
+		Quantity:            sl.Int(1),
+		Hardware: []datatypes.Hardware{{
+			Hostname: sl.String(d.Get("hostname").(string)),
+			Domain:   sl.String(d.Get("domain").(string)),
+		},
+		},
+		Location:  sl.String(strconv.Itoa(*dc.Id)),
+		PackageId: pkg.Id,
+		Prices: []datatypes.Product_Item_Price{
+			server,
+			os,
+			ram,
+			portSpeed,
+			priIpAddress,
+			pri_ipv6_addresses,
+			remoteManagement,
+			vpnManagement,
+			monitoring,
+			notification,
+			response,
+			vulnerabilityScanner,
+		},
+	}
+
+	// Add optional price ids.
+	// Add public bandwidth
+
+	if publicBandwidth, ok := d.GetOk("public_bandwidth"); ok {
+		publicBandwidthStr := "BANDWIDTH_" + strconv.Itoa(publicBandwidth.(int)) + "_GB"
+		bandwidth, err := getItemPriceId(items, "bandwidth", publicBandwidthStr)
+		if err != nil {
+			return datatypes.Container_Product_Order{}, err
+		}
+		order.Prices = append(order.Prices, bandwidth)
+	}
+
+	// Add prices of disks.
+	var arrayDrives []interface{}
+	var arrayDrivestemp []string
+	arrayDrivestemp = append(arrayDrivestemp, "HARD_DRIVE_1_00_TB_SATA_III")
+	arrayDrivestemp = append(arrayDrivestemp, "HARD_DRIVE_2_00TB_SATA_II")
+
+	for _, val := range arrayDrivestemp {
+		arrayDrives = append(arrayDrives, val)
+	}
+
+	diskLen := len(arrayDrives)
+	if diskLen > 0 {
+		for i, disk := range arrayDrives {
+			diskPrice, err := getItemPriceId(items, "disk"+strconv.Itoa(i), disk.(string))
+			if err != nil {
+				return datatypes.Container_Product_Order{}, err
+			}
+			order.Prices = append(order.Prices, diskPrice)
+		}
+	}
+
+	// Add storage_groups for RAID configuration
+	//hard coded support for disk controller, just RAID 1 is supported in this release of the resource
+	diskController, err := getItemPriceId(items, "disk_controller", "DISK_CONTROLLER_RAID_1")
```

Comment:
Should this be exposed in schema and be configurable?

Reply:
Added this note into the documentation: RAID_CONTROLLER - (reference) this version support a single RAID 1 controller pre configured with two hard drives 1 TB SATA drives, the idea is to keep the Network Gateway with minimal internal storage configuration since it’s primary purpose is to serve the network